Tests/iaas/flavor-naming/flavor-form-server.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+# adapted from https://github.com/python/cpython/blob/3.12/Lib/http/server.py
+from http.server import CGIHTTPRequestHandler, BaseHTTPRequestHandler, ThreadingHTTPServer
+from http.server import _get_best_family, _url_collapse_path
+import argparse
+import contextlib
+import os
+import socket
+import sys
+
+
+class RequestHandler(CGIHTTPRequestHandler):
+
+    def is_cgi(self):
+        collapsed_path = _url_collapse_path(self.path)
+        if not collapsed_path.startswith('/cgi-bin/'):
+            return False
+        self.cgi_info = '', collapsed_path[8:]
```

Comment:
```suggestion
        self.cgi_info = '', collapsed_path.removeprefix('/cgi-bin')
```
(that's [new in Python 3.9](https://docs.python.org/3/library/stdtypes.html#str.removeprefix))